pact/Pact/Core/Serialise.hs

```diff
@@ -112,30 +126,54 @@ serialisePact = PactSerialise
                            V1_CBOR -> V1.decodeRowData
                        )
   }
-  where
-    docEncode :: (a -> ByteString) -> a -> ByteString
-    docEncode enc o = toStrictByteString (encodeVersion V1_CBOR <> S.encodeBytes (enc o))
-
-    docDecode :: ByteString -> (DocumentVersion -> ByteString -> Maybe a) -> Maybe (Document a)
-    docDecode bs dec = case deserialiseFromBytes (liftA2 (,) decodeVersion S.decodeBytes) (fromStrict bs) of
-      Left _ -> Nothing
-      Right (_, (v,c)) ->  Document v <$> dec v c
 
 gEncodeRowData :: RowData -> GasM (PactError i) b ByteString
 gEncodeRowData rd = do
   encodedRow <- V1.encodeRowData rd
   pure $ toStrictByteString $ encodeVersion V1_CBOR <> S.encodeBytes encodedRow
 
+liftReplBuiltin :: ModuleData CoreBuiltin a -> ModuleData ReplCoreBuiltin a
+liftReplBuiltin = \case
+  ModuleData em ed -> let
+    defs' = over (traverseDefTerm . termBuiltin) RBuiltinWrap <$> _mDefs em
+    ed' = over (traverseDefTerm . termBuiltin) RBuiltinWrap <$> ed
+    in ModuleData (em{_mDefs = defs'}) ed'
+  InterfaceData im ed -> let
+    ifdefs = over (traverseIfDefTerm . termBuiltin) RBuiltinWrap <$> _ifDefns im
+    ed' = over (traverseDefTerm . termBuiltin) RBuiltinWrap <$> ed
+    in InterfaceData (im{_ifDefns = ifdefs}) ed'
+
+
 serialisePact_repl_spaninfo :: PactSerialise ReplCoreBuiltin SpanInfo
 serialisePact_repl_spaninfo = serialisePact
-  { _encodeModuleData = V1.encodeModuleData_repl_spaninfo
-  , _decodeModuleData = fmap LegacyDocument . V1.decodeModuleData_repl_spaninfo
+  { _encodeModuleData = docEncode V1.encodeModuleData_repl_spaninfo
```

Comment:
Oh man, seems like we missed the infrastructure for repl tests.

Reply:
We did, but now we have it 👍